### ISSUE TITLE
Some detail about installing Solargraph?

### ISSUE THREAD

**Issue by DcsMarionDickten** [tommasongr/nova-rails#12]
Maybe it's a dumb request. But I'm having the same issue with VS Code. Rails support presupposes Solargraph. But that's a Gem. Gems are installed on a pro project base. While I'm a pretty seasoned Rails developer by now I still don't understand how to install a Gem that is used for an editor. I am working on five Rails projects at the moment, with different versions of Ruby and Rails (I am supporting old projects, that's why). Now do I have to install Solargraph in all of them? Everybody discourages installing the gem using sudo too. Could you just insert one line in your readme at Nova that says "if you don't have Solargraph yet, please install it like so"?

**Comment by tommasongr** [comment]
Hi there! Don't worry, there are no dumb questions.

Just so you know Rails support does not presuppose Solargraph. For example I personally use this extension mostly for the commands, shortcuts, sidebars, etc. Solargraph is installed on my machine just for testing porpuses but is deactivated in Nova. The reason is I don't think is much useful since the "Rails magic" behind the scenes prevents Solargraph from being very smart. IMO of course.

Anyway gems are not actually installed on per project base. To do that you use Bundler, but if you run the command `gem install solargraph` the gem is installed globally. In fact this is the way they suggest to install the package [https://solargraph.org/guides/getting-started](https://solargraph.org/guides/getting-started). The only caveat is if you use something like rbenv for managing multiple ruby versions. In that case you have to install the gem for each version of ruby that you use.

Although, installing Solargraph is not enough for having it working with Rails. There are additional steps you have to follow. [This guide](https://www.youtube.com/watch?v=mkLDPpDSWng) is the best I've found and the process is the same for Nova.

I'll definitely update the README in a future release with at least this two links.

Thanks for pointing out this issue! I hope I answer your question.

**Comment by DcsMarionDickten** [comment]
Hi,

thank you so much for your answer.

It's always frustrating to have editors with extensions without proper documentation. That's one reason why I don't like VS Code. I run it with practically no plugins for that reason. I've just rechecked the Rails plugin page in VS Studio. It shows lots of pretty animated pictures of what it can do and zero information about how to set it up. Making me watch a video with advertising in it is already annoying, but not pointing me to it is really uncool.

So yes, it would be really cool if you put in a sentence like "If you want the full functionality, install Solargraph for each Ruby version that you need in your projects" and maybe supply the link too.

And if I may ask another question as I'm here:

I have got some before_action filters, for instance: 
	
	before_action :set_german_language

My Nova installation as I have set it up right now is not able to jump to the definition of set_german_language (which is in the same file). Is this something your extension could add? Because AFAIK this is a pretty common Rails thing.

So far I'm really happy with Nova, I'd love finally getting off RubyMine :-)

Best wishes

Marion Dickten


--------------------
Marion Dickten
DCS-Fürth - DIE ENTWICKLER
Badstr. 13-15
90762 Fürth

fon: +49 911 148899-15
mail: ***@***.***
web: www.dcs-fuerth.de

**Comment by tommasongr** [comment]
You're welcome!

Yes, I'll definitely work on improving the docs as you rightly suggested.

The before_action thing is exactly one of the missing capabilities I was talking about. Long story short is not possible to do that since those filters are not real method references until runtime. [Here](https://github.com/castwide/vscode-solargraph/issues/127#issuecomment-503764322) is the full explanation from the team. So if even Solargraph can't support it, I doubt I'd be able to.

**Comment by DcsMarionDickten** [comment]
I looked at the link you provided, and it just goes to show how little I know about implementing a programming environment. I kept thinking "why, it's no trouble, just check whether there's a macro in front of the symbol, and if so, check the indexed methods." BTW, I absolutely agree with njyrbecker about the relative importance of code graphing features.

Thank you for your response & successful coding!
